cmd/collector/app/http_handler.go

```diff
@@ -66,7 +67,8 @@ func (aH *APIHandler) saveSpan(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	contentType := r.Header.Get("Content-Type")
+	contentType, _, _ := mime.ParseMediaType(r.Header.Get("Content-Type"))
```

Comment:
one issue here (and in all similar statements)  is that the parsing error is swallowed, and will usually cause a later, less intelligible, error like `Unsupported content type: ` (i.e. nothing at the end). The preferred way to handle this is
```go
contentType, _, err := mime.ParseMediaType(r.Header.Get("Content-Type"))
if err != nil {
		http.Error(w, fmt.Sprintf("Cannot parse content type: %v", err), http.StatusBadRequest)
		return
}
```
it will require more failure tests to maintain the code coverage.

Reply:
Cool, so to you, it's valuable that parse errors here are surfaced as discrete new errors for Jaeger? Like, you want it such that the caller gets notified how their content type is wrong? ( Rather than just, "it's wrong" )

Reply:
( Because the latter seems already accounted for in the existing code )

Reply:
nm, re-read your comment and that's exactly what you want. Cool, I'll add.

Reply:
Every time the code swallows an error, someone gets burnt by it, and a fairy dies. It's always better to pass the full error to the user than to tell them "you f-ed up, no comment".